src/inc/til/env.h

```diff
@@ -0,0 +1,575 @@
+                                // If we did, pull it off.
+                                strip_trailing_null(data);
+
+                                if (!data.empty())
+                                {
+                                    if (is_path_var(valueName))
+                                    {
+                                        concat_var(valueName, std::wstring{ data });
+                                    }
+                                    else
+                                    {
+                                        set_user_environment_var(valueName, std::wstring{ data });
+                                    }
+                                }
+                            }
+
+                            valueName.resize(maxValueNameSize);
+                            valueData.resize(maxValueDataSize);
+                            valueNameSize = maxValueNameSize;
+                            valueDataSize = maxValueDataSize;
+                            index++;
+                        }
+                        index = 0;
+                    }
+                }
+            }
+        }
+
+        std::wstring expand_environment_strings(std::wstring input)
+        {
+            std::wstring expanded;
+            expanded.reserve(input.size());
+            bool isInEnvVarName = false;
+            std::wstring currentEnvVarName;
+            for (const auto character : input)
+            {
+                if (character == L'%')
+                {
+                    if (isInEnvVarName)
+                    {
+                        if (const auto envVarValue = _envMap.find(currentEnvVarName); envVarValue != _envMap.end())
+                        {
+                            expanded.append(envVarValue->second);
+                        }
+                        else
+                        {
+                            expanded.push_back(L'%');
+                            expanded.append(currentEnvVarName);
+                            expanded.push_back(L'%');
+                        }
+                        isInEnvVarName = false;
+                        currentEnvVarName.clear();
+                    }
+                    else
+                    {
+                        isInEnvVarName = true;
+                    }
+                }
+                else
+                {
+                    if (isInEnvVarName)
+                    {
+                        currentEnvVarName.push_back(character);
+                    }
+                    else
+                    {
+                        expanded.push_back(character);
+                    }
+                }
+            }
+            if (isInEnvVarName)
+            {
+                expanded.push_back('%');
+                expanded.append(currentEnvVarName);
+            }
+            return expanded;
+        }
+
+        void set_user_environment_var(std::wstring var, std::wstring value)

```

Comment:
Both `check_for_temp` and `expand_environment_strings` make copies of their arguments so both `var` and `value` can be a `std::wstring_view` instead.

Reply:
I've addressed all the comments except this one.

Switching to `wstring_view` means both `expand_environment_strings()` and `save_to_map()` need to change. I can do that if you like - but I'm unsure what implications that will have on the underlying map...

![image](https://user-images.githubusercontent.com/5821575/222241388-86214fa3-d7ff-420e-87ce-c0aa432cf1ca.png)
![image](https://user-images.githubusercontent.com/5821575/222241558-379e8965-ab1d-4696-8008-99284576126d.png)

Reply:
I'm an idiot - never mind - done.